packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx

```diff
@@ -293,6 +293,7 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
                 handleTabKey={ FocusZoneTabbableElements.all }
               >
                 <ul
+                  role={ shouldFocusOnContainer ? undefined : 'menu' }
```

Comment:
No, we should always be setting role = menu... This is always a menu no matter how the menu was brought up

Reply:
And why is being added here and not on the element that is the whole menu (e.g. the element that had the role before)

Reply:
Feel free to play around with this yourself. If we add role='menu' when the user has activated the contextual menu via keyboarding to the menu container, then the aria label on the menu IS NOT read. This change ensures that role='menu' always appears somewhere on the menu (as above on the menu item container we add it when focus is on the entire container.)

Reply:
I do not want to make a change that breaks reading the aria label on the entire menu for the keyboarding case.

Reply:
I just played with it and if a menu has an aria-label or an aria-labelledby it is read out when the menu is focused. Which is expected behavior. Are you not using the latest Edge/Narrator? We shouldn't make the content less accessible as a hack for IE (and/or an old narrator).

Reply:
We should always code against the standards, not against potentially faulty implementations of interpreting the standards (by a browser and/or screen reader)

Reply:
Okay, that sounds like a narrator bug, not one that we should hack around

Reply:
Fine. I'll remove it and always set role='menu'. How can I file a bug on the Narrator team? 

Even if the markup now is more "correct", the Edge + Narrator experience is now wrong when focus is on the first menu item.